eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -215,28 +216,52 @@ case class Color(r: Byte, g: Byte, b: Byte) {
 // @formatter:off
 sealed trait NodeAddress { def socketAddress: InetSocketAddress }
 sealed trait OnionAddress extends NodeAddress
+sealed trait IPAddress extends NodeAddress
+// @formatter:on
+
 object NodeAddress {
   /**
-    * Creates a NodeAddress from a host and port.
-    *
-    * Note that non-onion hosts will be resolved.
-    *
-    * We don't attempt to resolve onion addresses (it will be done by the tor proxy), so we just recognize them based on
-    * the .onion TLD and rely on their length to separate v2/v3.
-    */
+   * Creates a NodeAddress from a host and port.
+   *
+   * Note that non-onion hosts will be resolved.
+   *
+   * We don't attempt to resolve onion addresses (it will be done by the tor proxy), so we just recognize them based on
+   * the .onion TLD and rely on their length to separate v2/v3.
+   */
   def fromParts(host: String, port: Int): Try[NodeAddress] = Try {
     host match {
       case _ if host.endsWith(".onion") && host.length == 22 => Tor2(host.dropRight(6), port)
       case _ if host.endsWith(".onion") && host.length == 62 => Tor3(host.dropRight(6), port)
-      case _  => InetAddress.getByName(host) match {
+      case _ => InetAddress.getByName(host) match {
         case a: Inet4Address => IPv4(a, port)
         case a: Inet6Address => IPv6(a, port)
       }
     }
   }
+
+  def extractIPAddress(socketAddress: InetSocketAddress): Option[IPAddress] = {
+    val address = socketAddress.getAddress
+    address match {
+      case address: Inet4Address => Some(IPv4(address, socketAddress.getPort))
+      case address: Inet6Address => Some(IPv6(address, socketAddress.getPort))
+      case _ => None
+    }
+  }
+
+  private def isPrivate(address: InetAddress): Boolean = address.isAnyLocalAddress || address.isLoopbackAddress || address.isLinkLocalAddress || address.isSiteLocalAddress
```

Comment:
Have you find this recipe somewhere or did you take all `is*Local*` methods?

Reply:
I just looked at all the `is*Local*` methods exposed...
I guess the first step is to ship that and see if that works in the real-world or not (this feature is only informative for now, we don't act on it yet).